### PR TITLE
Fix style inconsistencies in Bash code

### PR DESCRIPTION
**Pull request by UniqMartin**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

We have accumulated quite a few style inconsistencies in our Bash code. I fixed all of them following the rules below that I inferred them from what I believe to be our preferred and/or predominant style. (Of course all of those rules are up for debate.)

- **Whitespace**
  - Indent by two spaces.
  - Align single-line items in `case` statements.
  - Put one space between single-line items in `case` and terminating `;;`.
  - Indent items in `case` statements one step (unlike in Ruby code).
  - Use multi-line `if`/`elsif`-`then` and `for`/`while`-`do` statements.
  - Omit space between function name and `()`.
  - Put function “signature” and opening curly brace on same line, e.g. `foo() {`.
  - Put a newline between shebang and shell code.
  - Omit space after redirection operators (`>`, `>>`, `<`), e.g. `>/dev/null`.
- **Redundant Stuff**
  - Omit unneeded trailing `;`.
  - Omit unneeded `-` in `<<-EOS` (we don't tab-indent our here documents).
  - Omit unneeded line continuation (`\`) in arrays.
- **Miscellaneous**
  - Prefer double-quoted strings over single-quoted strings if equally clear. (Continue to use single-quoted string if double-quoted string requires heinous escaping.)
  - Prefer `[[` over `[` and `test`.
  - Prefer `=` over `==` in `[[`.
  - Always quote variable usage and shell substitution unless word-splitting behavior is intended. (For consistency, do this also in assignments where special rules apply.)

### CODE REVIEW

**Comment by xu-cheng** [comment]
👍

File: Library/Homebrew/shims/super/ant [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
+
 export HOMEBREW_CCCFG="O$HOMEBREW_CCCFG"
+
 ant=/usr/bin/ant
-[ -x "$ant" ] || ant="$(${HOMEBREW_BREW_FILE} --prefix ant)/bin/ant"
+[[ -x "$ant" ]] || ant="$("$HOMEBREW_BREW_FILE" --prefix ant)/bin/ant"
```
Comment (xu-cheng) [inline Library/Homebrew/shims/super/ant]: Maybe we should use `HOMEBREW_PREFIX` directly.

File: Library/Homebrew/shims/super/mig [hunk 2]
```diff
@@ -1,3 +1,4 @@
 #!/bin/bash
+
 pwd="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
```
Comment (xu-cheng) [inline Library/Homebrew/shims/super/mig]: It seems that there are multiple styles for resolving the current directory. I'm wondering should we make them consistence as well?

File: Library/Homebrew/shims/super/ant (hunk 1 above)
Comment (UniqMartin) [inline Library/Homebrew/shims/super/ant]: True. Note that I avoided all changes to functionality, however subtle, and focused on the style issues only, so not addressing this was intentional.

File: Library/Homebrew/shims/super/mig (hunk 2 above)
Comment (UniqMartin) [inline Library/Homebrew/shims/super/mig]: Like in the other comment, I tend to agree. (But as before, not changing functionality and focusing on style changes only was intentional.)

**Comment by MikeMcQuaid** [comment]
👍 but if this fixes all syntax failures in our bash code: please consider adding a check for this to `brew test-bot` or `brew tests` or similar so they don't just break again.

**Comment by UniqMartin** [comment]
> but if this fixes all syntax failures in our bash code: please consider adding a check for this to `brew test-bot` or `brew tests` or similar so they don't just break again.

No, all the changes in this PR are purely about style. If there were any syntax errors, we would have been in more trouble (unless it's in a rarely used `superenv` shim).

However, your comment made me remember `test/test_bash.rb` and I pushed another commit to make sure it checks **all** our Bash code for syntax errors, including `superenv` shims and newly added or relocated code. (I think this test was added when `bin/brew` was still a huge monolithic piece of code and neither `brew.sh` nor `utils{,/*}.sh` existed yet.) Thanks for this! ❤️

**Comment by MikeMcQuaid** [comment]
Sorry, to clarify: I mean if you've fixed all style warnings it'd be great to push an automated `shellcheck` check somewhere. Thanks for that 👍

**Comment by UniqMartin** [comment]
I see. No, we're not quite there yet and most stuff I changed in this PR isn't the kind that would be pointed out by `shellcheck`, though I made sure not to introduce any new warnings. I'll make sure to put this on my TO DO list, but IMO that's definitely something for a follow-up PR.

**Comment by MikeMcQuaid** [comment]
I'd love to see Homebrew/brew "clean" enough at some point that we can have Shellcheck and RuboCop run by CI to check for regressions. I guess we could do some of that automatically if we have CI use a different set of rules and try to fix one set at a time?

**Comment by UniqMartin** [comment]
I fully agree and the idea how to do this is shaping up nicely in my head, thanks also to your input! The only thing that's holding me back from spending more time with this is that I'm currently trying to focus more on GSoC/Outreachy-related stuff and more pressing issues (and yes, this PR doesn't exactly prove my point). All those style-related tasks are definitely quite high on my agenda, if nobody beats me to it, and it would be nice to have some of that before 1.0.

**Comment by MikeMcQuaid** [comment]
@UniqMartin same here.

**Pull request closed without merging**